packages/dart/noports_core/lib/src/srv/relay_authenticators.dart

```diff
@@ -0,0 +1,238 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:at_chops/at_chops.dart';
+import 'package:at_client/at_client.dart';
+import 'package:mutex/mutex.dart';
+
+/// Clients which are authenticating to a relay may use a [RelayAuthenticator]
+/// to do so. Responsibility of a [RelayAuthenticator] is typically to
+/// - wait to receive some sort of challenge from the relay
+/// - send a response back to the relay based on that challenge (typically
+///   a signed payload)
+/// - wait for confirmation from the relay
+abstract interface class RelayAuthenticator {
```

Comment:
Made an interface for Authenticators, and moved the first implementation in here as RelayAuthenticatorLegacy